qkernel/src/syscalls/sys_proxy.rs

```diff
@@ -45,6 +46,209 @@ pub fn SysProxy(task: &mut Task, args: &SyscallArguments) -> Result<i64> {
+            }
+
+            parameters.para1 = &mut CacheConfig as *mut _ as u64;
+               
+            let ret = HostSpace::Proxy(
+                cmd,
+                parameters,
+            );
+
+            if ret == 0 {
+                task.CopyOutObj(&CacheConfig, args.arg1 as u64)?; 
+            }
+
+            return Ok(ret);
+        }
+        ProxyCommand::CudaDeviceGetLimit => {
+
+            let mut limit:usize = 0; 
+            
+            parameters.para1 = &mut limit as * mut _ as u64;
+
+            // let attribute: u32 = parameters.para2 as u32;
+            // error!("SysProxy CudaDeviceGetLimit, query about attribute: {}", attribute);
+
+            let ret = HostSpace::Proxy(
+                cmd,
+                parameters,
+            );
+            
+            if ret == 0 {
+                task.CopyOutObj(&limit, args.arg1 as u64)?; 
+            }
+
+            return Ok(ret);
+
+        }
+        ProxyCommand::CudaDeviceGetP2PAttribute => {
+            let mut value:i32 = 0;
+            parameters.para1 = &mut value as * mut _ as u64;
+
+            // let attribute: u32 = parameters.para2 as u32;
+            // error!("SysProxy CudaDeviceGetP2PAttribute, query about attribute: {}", attribute);
+
+            let ret = HostSpace::Proxy(
+                cmd,
+                parameters,
+            );
+
+            if ret == 0 {
+                task.CopyOutObj(&value, args.arg1 as u64)?;
+            }
+            return Ok(ret);
+
+        }
+        ProxyCommand::CudaDeviceGetPCIBusId => {
+            let mut pciBusIdAddress = CString::ToString(task, parameters.para1)?;
+
+            // error!("pciBusIdAddress: {}", pciBusIdAddress);
+
+            parameters.para1 = &mut pciBusIdAddress as * mut String as u64;
+
+            // error!("SysProxy CudaDeviceGetPCIBusId, query about device:{}", parameters.para3);
+
+            let ret = HostSpace::Proxy(
+                cmd,
+                parameters,
+            );
+
+           if ret == 0 {
+            task.CopyOutString(args.arg1, parameters.para2 as usize, &pciBusIdAddress)?;
+           }
+
+            return Ok(ret);
+        }
+        ProxyCommand::CudaDeviceGetSharedMemConfig => {
+            let mut sharedMemConfig:u32 = unsafe { *(parameters.para1 as *mut _ )};
+            // error!("value of sharedMemConfig is: {}", sharedMemConfig);
+
+            parameters.para1 = &mut sharedMemConfig as *mut _ as u64;
+
+            let ret = HostSpace::Proxy(
+                cmd, 
+                parameters,
+            );
+
+            if ret == 0 {
+                task.CopyOutObj(&sharedMemConfig, args.arg1 as u64)?;
+            }
+
+            return Ok(ret);
+
+        }
+        ProxyCommand::CudaDeviceGetStreamPriorityRange => {
+            let mut lowPriority:i32;
+            let mut highPriority:i32;
+
+            unsafe{
```

Comment:
I think this unsafe can be removed. Check how the code was done from ProxyCommand::CudaRegisterFunction?